From 242547ae3afade33ca18e876d129f34b47b646ae Mon Sep 17 00:00:00 2001
From: Eberhard Beilharz <eb1@sil.org>
Date: Tue, 14 Feb 2012 08:01:29 +0100
Subject: [PATCH] Xamarin-3418: Dispose menu items on Menu.Dispose

When disposing a menu it has to dispose all the menu items
it owns.

Change-Id: I0edb5564edfe0f5ee5a4066c68198082a2ea2cb4
---
 .../System.Windows.Forms/Menu.cs                   |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

Index: fieldworks-mono/mcs/class/Managed.Windows.Forms/System.Windows.Forms/Menu.cs
===================================================================
--- fieldworks-mono.orig/mcs/class/Managed.Windows.Forms/System.Windows.Forms/Menu.cs
+++ fieldworks-mono/mcs/class/Managed.Windows.Forms/System.Windows.Forms/Menu.cs
@@ -201,10 +201,17 @@
 		protected override void Dispose (bool disposing)
 		{		
 			if (disposing) {
+				if (menu_items != null) {
+					// MenuItem.Dispose removes the item from the list
+					while (menu_items.Count > 0) {
+						menu_items [0].Dispose ();
+					}
+				}
 				if (menu_handle != IntPtr.Zero) {
 					menu_handle = IntPtr.Zero;
 				}
 			}
+			menu_items = null;
 		}
 
 		// From Microsoft documentation is impossible to guess that 
